src/dra/7E4BC.c

```diff
@@ -1202,7 +1202,97 @@ void func_80123A60(Entity* entity) {
     DestroyEntity(entity);
 }
 
-INCLUDE_ASM("dra/nonmatchings/7E4BC", func_80123B40);
+void func_80123B40(Entity* self) {
+    Entity copy;
+    Primitive* prim;
+    s16 params;
+    s32 i;
+
+    params = self->params;
+    copy = PLAYER;
+
+    if (!self->step) {
+        self->animSet = PLAYER.animSet = 1;
+        self->animCurFrame = PLAYER.animCurFrame = 0x9F;
+        self->palette = PAL_OVL(0x10D);
+        self->unk5A = 0xC;
+        self->params = 0x2600;
+        EntityPlayerBlinkWhite(self);
+        self->params = params;
+
+        PLAYER = copy;
+
+        if (!(self->flags & 0x800000)) {
+            DestroyEntity(self);
+            return;
+        }
+
+        self->flags = 0x0C830000;
+        self->velocityY = FIX(-3.0);
+        SetSpeedX(-0x1AAAA);
+        self->ext.stub[0x28] = 0x80;
```

Comment:
@Xeeynamo Do we need to create an ET_XXX here?

Reply:
Yep 100%. The use of `self->ext.stub` should be disallowed and it's there just for padding

Reply:
Sounds good. I'll see if any of the others are relevant otherwise create a new one.

Reply:
We usually create new ones based on the context of the function. For example this one is calling `EntityPlayerBlinkWhite` with the same parameter. So maybe we can create a new struct called `ET_BlinkWhite blinkWhite` and use `self.ext.blinkWhite.unkA4`, or something like that.